pkg/generic/thriftidl_provider.go

```diff
@@ -50,6 +59,20 @@ func NewThriftFileProvider(path string, includeDirs ...string) (DescriptorProvid
 	if err != nil {
 		return nil, err
 	}
+
+	useDynamicgoConversion := os.Getenv(UseDynamicgoConv) == "True"
+	if useDynamicgoConversion {
```

Comment:
encapsulate the construction for ServiceDescriptor as a function is better, there are five places in this file that have similar construction code

Reply:
fixed